src/platforms/hosted/serial_unix.c

```diff
@@ -43,6 +47,84 @@ static uint8_t read_buffer[READ_BUFFER_LENGTH];
 static size_t read_buffer_fullness = 0U;
 static size_t read_buffer_offset = 0U;
 
+#ifndef _WIN32
+inline int closesocket(const int socket)
+{
+	return close(socket);
+}
+#endif
+
+/* Socket code taken from https://beej.us/guide/bgnet/ */
+static bool try_opening_network_device(const char *const name)
+{
+	if (!name)
+		return false;
+
+	// Maximum legal length of a hostname
+	char hostname[256];
+
+	// The service name or port number
+	char *service_name;
```

Comment:
Please move the definition of this variable down to where it gets initialised on line 85.